src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights.AspNetCore/TargetingTelemetryInitializer.cs

```diff
@@ -37,27 +36,23 @@ protected override void OnInitializeTelemetry(HttpContext httpContext, RequestTe
                 throw new ArgumentNullException(nameof(telemetry));
             }
 
+            if (telemetry is not ISupportProperties telemetryWithSupportProperties)
+            {
+                return;
+            }
+
             if (httpContext == null)
             {
                 throw new ArgumentNullException(nameof(httpContext));
             }
 
-            // Extract the targeting id from the http context
-            string targetingId = null;
+            //
+            // Extract the targeting info from the http context
+            TargetingContext targetingContext = httpContext.Items[DefaultHttpTargetingContextAccessor.TargetingContextLookup] as TargetingContext;
```

Comment:
Is it intentional to let this throw if the cache doesn't exist? If so, we should throw something more meaningful to users.

Reply:
My undersatnding is that the `as` casting won't throw- it'll either give the requested object or return null. From https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/type-testing-and-cast?redirectedfrom=MSDN#as-operator

I might be ignoring different versions or something where that's the case.

Reply:
I was talking about `httpContext.Items[DefaultHttpTargetingContextAccessor.TargetingContextLookup]`. Will it throw if the cache doesn't exist?

Reply:
Oh thank you- you're right. Updated